test/e2e/mco_test.go

```diff
@@ -27,3 +31,63 @@ func TestClusterOperatorRelatedObjects(t *testing.T) {
 		t.Error("ClusterOperator.RelatedObjects should contain the MCO namespace")
 	}
 }
+
+func TestMastersSchedulable(t *testing.T) {
+	cs := framework.NewClientSet("")
+	schedulerCR, err := cs.ConfigV1Interface.Schedulers().Get("cluster", metav1.GetOptions{})
+	if err != nil {
+		t.Fatal("Error while listing scheduler CR")
+	}
+	schedulerCR.Spec.MastersSchedulable = true
+	if _, err = cs.ConfigV1Interface.Schedulers().Update(schedulerCR); err != nil {
+		t.Fatalf("Error while updating scheduler CR with error %v", err)
+	}
+	err = waitForAllMastersUpdate(cs, t, true)
+	if err != nil {
+		t.Fatalf("Expected all master nodes to be schedulable but it's not the case with %v", err)
+	}
+	// Reset scheduler CR
+	schedulerCR, err = cs.ConfigV1Interface.Schedulers().Get("cluster", metav1.GetOptions{})
+	if err != nil {
+		t.Fatal("Error while listing scheduler CR")
+	}
+	schedulerCR.Spec.MastersSchedulable = false
+	if _, err = cs.ConfigV1Interface.Schedulers().Update(schedulerCR); err != nil {
+		t.Fatalf("Error while updating scheduler CR with error %v", err)
+	}
+	err = waitForAllMastersUpdate(cs, t, false)
+	if err != nil {
+		t.Fatalf("Expected all master nodes to be unschedulable but it's not the case with %v", err)
+	}
+}
+
+func toggleMasterNodesSchedulability(cs *framework.ClientSet, masterSchedulable bool, t *testing.T) bool {
```

Comment:
question: this isn't really toggling the actual schedulableness of master it's just checking if whether or not it's correctly set no?

Reply:
Yeah, you're right. I initially wanted the toggling to happen here instead of https://github.com/openshift/machine-config-operator/pull/937/files#diff-737b90e6d061aba6fa31500eba42868eR45 but I kind of moved away. I will change it `checkAllMastersSchedulable` or something like that.